potodo/_github.py

```diff
@@ -1,32 +1,45 @@
 import requests
 
-from subprocess import getoutput
+from subprocess import check_output
 
 
-def get_repo_url():
-    url = getoutput("git remote get-url --all upstream")
+def get_repo_url() -> str:
+    """
+    Tries to get the repository url from git commands
+    """
+    url: str = str(check_output("git remote get-url --all upstream"))
     if "fatal" in url:
-        url = getoutput("git remote get-url --all origin")
+        url = str(check_output("git remote get-url --all origin"))
     if "fatal" in url:
+        # If the commands didn't work
         raise ValueError(
             f"Unknown error. `git get-url --all upstream|origin` returned {url}"
         )
     return url
 
 
-def get_repo_name():
+def get_repo_name() -> str:
+    """
+    Will get the repository url from git commands then remove useless stuff to get ORG/NAME
+    """
     repo_url: str = get_repo_url()
-    repo_url = repo_url.replace("https://github.com/", "")
-    repo_url = repo_url.replace("git@github.com:", "")
-    repo_name = repo_url.replace(".git", "")
+    # Removes useless stuff. If it isn't there then nothing happens
+    repo_url = repo_url.strip("https://github.com/")
+    repo_url = repo_url.strip("git@github.com:")
+    repo_name = repo_url.strip(".git")
     return repo_name
 
 
-def get_reservation_list():
+def get_reservation_list() -> dict:
```

Comment:
Better not be that strict about the types, to stil alow for some duck typing, just be strict about the protocol the returned value should implement, maybe use `Mapping` here, even better `Mapping[str, str]`.

Reply:
I don't understand why 🤔

Reply:
Two reasons, one for `Mapping` and one for `[str, str]`

# Duck typing

Imagine you keep the dict here, you'll end up with a variable of type `dict` getting the result of your function, it's OK.
Imagine you create another function `accepting` the `dict` as a parameter, you'll type it as `dict` as it's a dict, long way to the issue, but still almost OK.
Now I'm using you code, I want to call your other function (the one only accepting `dict`) but I'm having an `OrderedDict`: I can't call your function, yet the function would properly work with it.

Calling a function with an implementation that has not been expected by the implementer, and yet having something that work, is the good part of duck typing. By strictly enforcing `dict` here, you're disallowing the nice part of duck typing.

By using `Mapping` instead, you're not saying you "want that" but "want something that implement __len__, __iter__, __getitem__ and so on. So if I give you an OrderedDict or any "dict-like" implementation of my own, it will still work.

So OK, OK, returning `dict` is not that bad, as it's true you're returning a dict, but telling a parameter is of type dict is bad, and I fear if you say you return `dict` it will ... propagate on the other parts of the code and we'll soon find `dict` as parameters (yes there's already a few).

# Generics

dict is not generic, `typing.Mapping` is, this mean you can express what's in the dict, like `Mapping[str, str]`, so type checkers would be able to whine if you do `my_mapping["foo"] = my_mapping.get("bar")`, because `.get` can return None, and you said it contains only strings.

Also it's a nice information for readers of your code: they don't only learn you're returning a mapping, they learn you're returning a mapping of strings to strings, so if they do dat["foo"] they'll get a string, not a None, not an int...

Reply:
Thanks ! It's fixed